### ISSUE TITLE
Wrong types are used in packets starting in 1.20.2

### ISSUE THREAD

**Issue by MrKinau** [Pokechu22/Burger#49]
After comparing 1.20.1 and 1.20.2 changes it looks like all `writeByte` and `writeShort` method calls are now `writeVarInt`, which is not the case. Additionally `writeDouble` is called `writePosition`, although it's still a `writeDouble` call. Maybe it's a burger issue?

![image](https://github.com/Pokechu22/BurgerWebViewer/assets/13185260/0b797381-3e27-4b53-8660-a573e0a63e3f)

**Comment by Pokechu22** [comment]
Based on https://pokechu22.github.io/Burger/diff_1.20.1_1.20.2.json and https://pokechu22.github.io/Burger/diff_1.20.1_1.20.2.html#packets:play_clientbound_01 this is a Burger issue.

I vaguely recall we treat any unknown type as `position`, but that still doesn't make sense...

**Comment by Pokechu22** [comment]
Definitely a burger issue, and it's a very old assumption that broke. Here's what `PLAY_CLIENTBOUND_02` looks like in 1.20.1 (ut.class):

```java
  public void a(sf $$0) {
    $$0.d(this.a);
    $$0.writeDouble(this.b);
    $$0.writeDouble(this.c);
    $$0.writeDouble(this.d);
    $$0.writeShort(this.e);
  }
```

and here's what it looks like in 1.20.2 (wy.class):

```java
  public void a(so $$0) {
    $$0.c(this.a);
    $$0.a(this.b);
    $$0.a(this.c);
    $$0.a(this.d);
    $$0.l(this.e);
  }
```

Here's what the packet buffer looked like in 1.20.1:

```java
public class sf extends ByteBuf {
  // ...
  private final ByteBuf f;
  // ...
  public ByteBuf writeShort(int $$0) {
    return this.f.writeShort($$0);
  }
  // ...
  public ByteBuf writeDouble(double $$0) {
    return this.f.writeDouble($$0);
  }
  // ...
}
```

and here's what it looks like in 1.20.2:

```java
public class so extends ByteBuf {
  // ...
  private final ByteBuf d;
  // ...
  public so l(int $$0) {
    this.d.writeShort($$0);
    return this;
  }
  // ...
  public so a(double $$0) {
    this.d.writeDouble($$0);
    return this;
  }
  // ...
}
```

In older versions, calling `writeDouble` would return the inner `ByteBuf`, and then it wouldn't be possible to chain calls to other functions (e.g. `buffer.writeDouble(1.5).writeVarInt(4)` would fail with `writeVarInt` not existing). It looks like this flaw existed even back in 1.7.10 (when the netty re-write to use `ByteBuf` was first added), though I guess I didn't remember it. I don't think any MC code currently chains calls.

But, this breaks the assumption Burger makes that calls like `writeShort` won't be obfuscated - burger uses the name to determine the type (because `writeShort` takes an `int` as its parameter, not a `short`). Specifically, the assumptions is that anything with an `int` as a parameter is `writeVarInt`, and anything else that it doesn't recognize is `writePosition`.

This is definitely solvable, but it'll be a bit of a mess.

**Comment by Pokechu22** [comment]
Fixed by dad94c73745803e7c464608f13eaa04cc1373189.

<details><summary>(And regenerated for these versions in ecf75c16a757c9b4e72d265dd3d853c236698719):</summary>

```
1.7.10 1.8
1.8.9 1.9
1.8 1.10
1.8 1.8.9
1.8 1.9
1.9.4 1.10
1.10.2 1.11
1.11.2 1.12
1.12.2 1.13
1.13.2 1.14
1.14.4 1.15
1.15.2 1.16
1.16.5 1.17
1.17.1 1.18
1.18.2 1.19
1.19.4 1.20
1.20 1.20.1
1.20.1 23w31a
23w31a 23w32a
23w32a 23w33a
23w33a 23w35a
23w35a 1.20.2-pre1
1.20.2-pre1 1.20.2-pre2
1.20.2-pre2 1.20.2-pre3
1.20.2-pre3 1.20.2-pre4
1.20.2-pre4 1.20.2-rc1
1.20.2-rc1 1.20.2-rc2
1.20.2-rc2 1.20.2
1.20.1 1.20.2
1.20.2 23w40a
23w40a 23w41a
23w41a 23w42a
23w42a 23w43a
23w43a 23w43b
23w43b 23w44a
23w44a 23w45a
23w45a 23w46a
23w46a 1.20.3-pre1
1.20.3-pre1 1.20.3-pre2
1.20.3-pre2 1.20.3-pre3
1.20.3-pre3 1.20.3-pre4
1.20.3-pre4 1.20.3-rc1
1.20.3-rc1 1.20.3
1.20.2 1.20.3
1.20.3 1.20.4-rc1
1.20.4-rc1 1.20.4
1.20.3 1.20.4
1.20.4 23w51a
23w51a 23w51b
```

</details>
